lib/python/treadmill_aws/spot.py

```diff
@@ -0,0 +1,98 @@
+from datetime import datetime, timedelta
+from typing import Dict
+
+from treadmill_aws.awscontext import GLOBAL
+
+_CORES = {
+    "medium": 2,
+    "large": 2,
+    "xlarge": 4,
+    "2xlarge": 8,
+    "4xlarge": 16,
+    "10xlarge": 40,
+    "12xlarge": 48,
+    "16xlarge": 64,
+    "24xlarge": 96,
+}
+
+
+def cores_of(instance_type):
+    return _CORES[instance_type.split('.').pop()]
+
+
+def get_latest_price(types, az):
+    price_list = GLOBAL.ec2.describe_spot_price_history(
+        InstanceTypes=types,
+        AvailabilityZone=az,
+        StartTime=datetime.now() - timedelta(days=1),
+        EndTime=datetime.now(),
+        ProductDescriptions=["Linux/UNIX"]).get('SpotPriceHistory')
+    if price_list:
+        return price_list
+
+
+def get_pricing(instance_types, availability_zones):
+    price_list = dict.fromkeys(availability_zones, {})
+    for az in availability_zones:
+        price_list[az] = dict.fromkeys(instance_types, {})
+        for result in get_latest_price(instance_types, az):
```

Comment:
get_latest_price can return None (there is if price_list), then this will crash.

Reply:
Thanks Michael for spotting this :) I actually did not mean to check this module in yet, I included it by mistake. I wrote these sample functions as a demo. I might rewrite the exact function that the autoscaler needs. I needs a better algorithm that takes into account the event when a SIR fail